core/src/replay_stage.rs

```diff
@@ -8598,4 +8625,60 @@ pub(crate) mod tests {
         assert_eq!(reset_fork, Some(4));
         assert_eq!(failures, vec![HeaviestForkFailures::LockedOut(4),]);
     }
+
+    #[test]
+    fn test_tower_reload_missing() {
```

Comment:
nit: make tests name more descriptive that we are switching identity for tower here?